src/bin/elasticsearch_exporter.rs

```diff
@@ -0,0 +1,359 @@
+    )]
+    elasticsearch_skip_metrics: HashMapVec,
+
+    #[clap(long = "exporter_poll_default_interval_ms", default_value = "5000")]
+    exporter_poll_default_interval_ms: u64,
+
+    #[clap(long = "exporter_skip_zero_metrics")]
+    exporter_skip_zero_metrics: bool,
+
+    #[clap(long = "exporter_poll_intervals", default_value = "cluster_health=5s")]
+    exporter_poll_intervals: HashMapDuration,
+}
+
+#[derive(Debug, Clone, Default)]
+struct HashMapDuration(ExporterPollIntervals);
+
+impl FromStr for HashMapDuration {
+    type Err = SimpleError;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let mut map = ExporterPollIntervals::new();
+
+        let parts = input.trim().split("&").into_iter().collect::<Vec<&str>>();
+
+        for part in parts.into_iter() {
+            match part.split_once("=") {
+                Some((key, value)) => match value.parse::<humantime::Duration>() {
+                    Ok(time) => {
+                        let _ = map.insert(key.to_string(), *time);
+                    }
+                    Err(e) => {
+                        return Err(SimpleError(format!(
+                            "Failed to parse time for key {} value {} err {}",
+                            key, value, e
+                        )))
+                    }
+                },
+                None => {
+                    return Err(SimpleError(format!(
+                        "Usage `{}`, you provided `{}`",
+                        HASH_MAP_STR_FORMAT, part
+                    )))
+                }
+            }
+        }
+
+        Ok(Self(map))
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+struct HashMapVec(CollectionLabels);
+
+impl FromStr for HashMapVec {
+    type Err = SimpleError;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let mut map = CollectionLabels::new();
+
+        let parts = input.trim().split("&").into_iter().collect::<Vec<&str>>();
+
+        for part in parts.into_iter() {
+            match part.split_once("=") {
+                Some((key, value)) => {
+                    let labels = value
+                        .split(",")
+                        .map(|value| value.to_string())
+                        .collect::<Vec<String>>();
+
+                    if let Some(values) = map.get_mut(key) {
+                        values.extend(labels);
+                    } else {
+                        let _ = map.insert(key.to_string(), labels);
+                    }
+                }
+                None => {
+                    return Err(SimpleError(format!(
+                        "Usage `{}`, you provided `{}`",
+                        HASH_MAP_STR_FORMAT, part
+                    )))
+                }
+            }
+        }
+
+        Ok(Self(map))
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+struct HashMapStr(Labels);
```

Comment:
Same as with Opts, I'd move HashMapStr and SimpleError to their own files. Will be easier to test and maintain.